packages/api/lib/granule-remove-from-cmr.js

```diff
@@ -73,31 +70,46 @@ const unpublishGranule = async (
     }
   }
 
-  return knex.transaction(async (trx) => {
-    let pgGranule;
-    if (pgGranuleCumulusId) {
-      [pgGranule] = await granulePgModel.update(
-        trx,
-        {
-          cumulus_id: pgGranuleCumulusId,
-        },
-        {
-          published: false,
-          // using `undefined` would result in Knex ignoring this binding
-          // for the update. also, `undefined` is not a valid SQL value, it
-          // should be `null` instead
-          cmr_link: trx.raw('DEFAULT'),
-        },
-        ['*']
+  try {
+    return await knex.transaction(async (trx) => {
+      let pgGranule;
+      if (pgGranuleCumulusId) {
+        [pgGranule] = await granulePgModel.update(
+          trx,
+          {
+            cumulus_id: pgGranuleCumulusId,
+          },
+          {
+            published: false,
+            // using `undefined` would result in Knex ignoring this binding
+            // for the update. also, `undefined` is not a valid SQL value, it
+            // should be `null` instead
+            cmr_link: trx.raw('DEFAULT'),
+          },
+          ['*']
+        );
+      }
+      const dynamoGranule = await granuleDynamoModel.update(
+        { granuleId: granule.granuleId },
+        { published: false },
+        ['cmrLink']
       );
+      await _removeGranuleFromCmr(granule);
+      return { dynamoGranule, pgGranule };
+    });
+  } catch (error) {
```

Comment:
I don't have a follow-up here. I think it's right as-is and we're throwing the error.

Reply:
Well the `catch` could be reached from anything in the transaction throwing.

But you're making me realize that we only need to revert the granule model changes if the granule in Dynamo was actually updated. The way the code is written, it doesn't matter because it just reverts the granule in Dynamo to what it initially was no matter what throws the error, which is always desired in the case of any error but unnecessary if the dynamo update never happened. A small update here is probably warranted.

Thoughts?

Reply:
What about this?

https://github.com/nasa/cumulus/pull/2358/commits/6fd776361271648ed8b12234a358bd3f06acf4ac

Reply:
Yeah, good point. I think that works. And having the whole transaction wrapped in a `try` makes sense. We've talked this over before and it's the clearest way. It just always brings up the question for me "what are we actually expecting to throw?".

Can't wait to ditch Dynamo 😄